File: packages/java/parser-jvm-plugin-backbone/src/test/resources/com/vaadin/hilla/parser/plugins/backbone/iterable/expected/IterableEndpoint.ts
import { EndpointRequestInit as EndpointRequestInit_1 } from "@vaadin/hilla-frontend";
import type Foo_1 from "./com/vaadin/hilla/parser/plugins/backbone/iterable/IterableEndpoint/Foo.js";
import client_1 from "./connect-client.default.js";
async function getFooAnotherCustomIterable_1(init?: EndpointRequestInit_1): Promise<Array<Foo_1 | undefined> | undefined> { return client_1.call("IterableEndpoint", "getFooAnotherCustomIterable", {}, init); }
async function getFooArray_1(init?: EndpointRequestInit_1): Promise<Array<Foo_1 | undefined> | undefined> { return client_1.call("IterableEndpoint", "getFooArray", {}, init); }
async function getFooCustomIterable_1(init?: EndpointRequestInit_1): Promise<Array<Foo_1 | undefined> | undefined> { return client_1.call("IterableEndpoint", "getFooCustomIterable", {}, init); }
async function getFooIterable_1(init?: EndpointRequestInit_1): Promise<Array<Foo_1 | undefined> | undefined> { return client_1.call("IterableEndpoint", "getFooIterable", {}, init); }
async function getFooList_1(init?: EndpointRequestInit_1): Promise<Array<Foo_1 | undefined> | undefined> { return client_1.call("IterableEndpoint", "getFooList", {}, init); }
async function getFooSet_1(init?: EndpointRequestInit_1): Promise<Array<Foo_1 | undefined> | undefined> { return client_1.call("IterableEndpoint", "getFooSet", {}, init); }
async function getSpecializedIterable_1(init?: EndpointRequestInit_1): Promise<Array<string | undefined> | undefined> { return client_1.call("IterableEndpoint", "getSpecializedIterable", {}, init); }
async function getSpecializedIterableCustom_1(init?: EndpointRequestInit_1): Promise<Array<Foo_1 | undefined> | undefined> { return client_1.call("IterableEndpoint", "getSpecializedIterableCustom", {}, init); }
export { getFooAnotherCustomIterable_1 as getFooAnotherCustomIterable, getFooArray_1 as getFooArray, getFooCustomIterable_1 as getFooCustomIterable, getFooIterable_1 as getFooIterable, getFooList_1 as getFooList, getFooSet_1 as getFooSet, getSpecializedIterable_1 as getSpecializedIterable, getSpecializedIterableCustom_1 as getSpecializedIterableCustom };

File: packages/java/typescript-generator/src/test/java/com/vaadin/hilla/parser/plugins/backbone/iterable/IterableEndpoint.java
/*
 * Copyright 2000-2025 Vaadin Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package com.vaadin.hilla.parser.plugins.backbone.iterable;

import com.vaadin.hilla.parser.testutils.annotations.Endpoint;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashSet;
import java.util.Iterator;
import java.util.List;
import java.util.Set;

@Endpoint
public class IterableEndpoint {
    public AnotherCustomIterable<Foo> getFooAnotherCustomIterable() {
        return new AnotherCustomIterable<>();
    }

    public Foo[] getFooArray() {
        return new Foo[] {};
    }

    public CustomIterable<Foo> getFooCustomIterable() {
        return new CustomIterable<>();
    }

    public Iterable<Foo> getFooIterable() {
        return Arrays.asList(new Foo(), new Foo());
    }

    public SpecializedIterable getSpecializedIterable() {
        return new SpecializedIterable();
    }

    public SpecializedIterableCustom getSpecializedIterableCustom() {
        return new SpecializedIterableCustom();
    }

    public List<Foo> getFooList() {
        return new ArrayList<>();
    }

    public Set<Foo> getFooSet() {
        return new HashSet<>();
    }

    public static class AnotherCustomIterable<T> implements Iterable<T> {
        private final List<T> list = new ArrayList<>();

        @Override
        public Iterator<T> iterator() {
            return list.iterator();
        }
    }

    public static class CustomIterable<T> implements Iterable<T> {
        private final List<T> list = new ArrayList<>();

        @Override
        public Iterator<T> iterator() {
            return list.iterator();
        }
    }

    public static class SpecializedIterable extends ArrayList<String> {
    }

    public static class SpecializedIterableCustom extends ArrayList<Foo> {
    }

    public static class Foo {
        public String bar = "bar";
    }
}
